### PR TITLE
NEW Add separate frontend bundle and TextFilter component

### LINKED ISSUE #74: Front end Text filter
After #73 is done, create a filter type for a text filter.

Designs: https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/asset/components/5c09c9d18e1f85001158544f

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #74

### CODE REVIEW

File: client/src/components/Frontend/TextFilter.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      );
+    }
+
+    // Two or more columns selected
+    const lastColumn = columns[columns.length - 1];
+    const firstColumns = columns.slice(0, columns.length - 1).join(', ');
+    return i18n.inject(
+      i18n._t(
+        'CKANTextFilter.FILTER_BY_MANY',
+        'Search by {firstColumns} or {lastColumn}'
```
Comment (ScopeyNZ) [inline client/src/components/Frontend/TextFilter.js]: Oxford comma?

File: client/src/components/Frontend/tests/TextFilter-story.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ))
+  .add('Three columns', () => (
+    <div>
+      <TextFilter
+        id="4"
+        columns={['topic', 'organisation', 'department']}
+        TextFieldComponent={TextField}
+      />
+    </div>
+  ));
```
Comment (ScopeyNZ) [inline client/src/components/Frontend/tests/TextFilter-story.js]: ❤️

**Comment by robbieaverill** [comment]
**Hold** until we've clarified the design aspect

**Comment by robbieaverill** [comment]
I've updated this to use a provided label via props instead of a generated placeholder on the input field

File: client/src/components/Frontend/TextFilter.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+TextFilter.defaultProps = {
+  columns: [],
+  extraClass: '',
+  label: '',
+};
+
+export { TextFilter as Component };
+
+export default inject(
```
Comment (ScopeyNZ) [inline client/src/components/Frontend/TextFilter.js]: Do we want to use Injector on the front end components? Considering our advice will be "just copy the code and edit for customisations" I feel that keeping it simple is best.

Comment (robbieaverill) [reply client/src/components/Frontend/TextFilter.js]: Good point - I didn't even think about it. I guess we should just render inputs directly instead

Comment (ScopeyNZ) [reply client/src/components/Frontend/TextFilter.js]: I guess there's nothing to prevent us from having Reactstrap as a dependency though.

**Comment by robbieaverill** [comment]
Updated - I've also removed the "Frontend" separation of components to match a bit closer with #122. Whichever gets merged first will need to update the webpack config to match the other.

**Comment by robbieaverill** [comment]
Just realised that the story book doesn't make sense now that we've removed dynamic label generation - have updated it

**Comment by robbieaverill** [comment]
Replaced in #135

**Pull request closed without merging**